### DISCUSSION TITLE
Here's how long this takes

### DISCUSSION THREAD

**Discussion by brock** [TheLastGimbus/GooglePhotosTakeoutHelper#156] [Show and tell]
Logging my experience with the dart version so folks have an idea of CPU consumption, duration, etc.

* Running `./gpth-macos` on a 2019 MBP 2.3 GHz 8-Core Intel Core i9
* Selecting 151 zip files on a local NAS connecting over samba
* Each file is roughly 2GB
* Copying into folder on local NAS (samba)
* Using interactive mode, selecting the year/month folder structure

![Screenshot 2023-01-19 at 14 42 48](https://user-images.githubusercontent.com/654725/213555451-345d9bcd-b8fb-4404-9c7b-e1edbb0d92fb.png)

![Screenshot 2023-01-19 at 14 42 51](https://user-images.githubusercontent.com/654725/213555469-b327ac4e-e32a-4674-953a-7465556520db.png)

Started at 1135am, now at 1445 (3 hours) and just started zip number 15, so **roughly 12 minutes for each 2GB file**.

**Comment by TheLastGimbus** [comment]
Thanks for sharing! One small thing...

> just started zip number 15, so roughly 12 minutes for each 2GB file

Umm, so you're on the step of extracting zips? Here is *the thing*:

I was getting ton of emails like "how am i supposed to extract zip plz help", so i decided i will incorporate their extraction *into* gpth. Thing is, Dart is a very productive language, but not the fastest one - you can probably get better unzipping performance in dedicated stuff like 7zip.

More on, `gpth` does so without deleting any zips - meaning, after running it, you may have 2x original space used. So in your case, you will need 151x2GB **x 2** = 604GB of free space - if you have that much, then okay, but for someone that doesn't (and for performence stuff), i would recommend unziping/deleting zips one by one, and then playing with gpth with cli args

this will also be much better for big operations like this, because if gpth fails in any step, it will delete the unzipped folder, and will need to re-do all of that :100: 

(And, as i stated in the other thread, you can run this headless on NAS :sweat_smile: ) 

> roughly 12 minutes for each 2GB file

Huh... on my Lenovo Ideapad 5 (Ryzen 4700+NVMe disk) it took me one minute for **whole process**, for one 2.5gb zip with 1209 photos... Maybe it's all thanks to NVMe :shrug:

**Comment by wgcj** [comment]
Took less than 10 minutes for me, with 56 2gb files pre-extracted with unzip '*.zip'
